SPEC.md

```diff
@@ -7957,7 +7996,7 @@ File join_paths(File, Array[String]+)
 File join_paths(Array[String]+)
 ```
 
-Joins together two or more paths into an absolute path in the host filesystem.
+Joins together two or more paths into an absolute path in the execution environment's filesystem.
```

Comment:
We should define these somewhere. Execution environment can mean multiple different things in different contexts. IE in this case it could be
- the trusted compute environment ie a cloud project
- a VM
- The running docker container